noir-projects/noir-protocol-circuits/crates/rollup-lib/src/block_root/block_root_rollup_inputs.nr

```diff
@@ -49,13 +49,12 @@ pub struct BlockRootRollupInputs {
     prover_id: Field,
     // Fields for blob verification made of all tx effects in this block
     // (will likely change to be accumulated and sent up to the final root)
-    // TODO(Miranda): Temporarily working with one blob to ensure things run
-    blob_fields: [Field; FIELDS_PER_BLOB],
+    blob_fields: [Field; FIELDS_PER_BLOB * BLOBS_PER_BLOCK],
     // see blob/src/main.nr -> main() - this is used for creating the challenge z
-    blob_commitment: [Field; 2],
-    // The EVM blob hash, can be injected here as the contract check its validity vs the blob_public_inputs below
-    // NB: to fit it into a field, we remove the first byte denoting the VERSIONED_HASH_VERSION_KZG
-    blob_hash: Field,
+    blob_commitments: [[Field; 2]; BLOBS_PER_BLOCK],
+    // Flat sha256 hash of the EVM blob hashes, can be injected here as the contract check its validity vs the blob_public_inputs below
+    // NB: to fit it into a field, we truncate to 31 bytes
```

Comment:
Todo: We should have a think about what happens if Ethereum updates blobs, to introduce a new "version" of the versioned hash. Suppose we end up in a world with more than one value for a "first byte" is possible for a versioned-blob-hash. And suppose Aztec can't update at the exact same pace as Ethereum. Are there any attacks that can be done, if we're omitting this first byte?

Reply:
From only a quick think, I believe the answer is no. Simply because our repo-coded version = `0x01`, so if the version updates, ours will be out of sync, and anything going on-chain will fail. Only the circuits deal with the truncated version of the hash, and they 'don't care' what the version is, just to have a number to hash into the header.
Since this 'version' number is one constant on the EVM, i don't think they will ever allow two versions to be running at once